File: app/models/player.rb
class Player < ActiveRecord::Base
  belongs_to :user
  belongs_to :table

  validates :user, :presence => true
  validates :user_id, :uniqueness => { :scope => :table_id }
  validates :table, :presence => true
  validates :direction, :presence => true, :inclusion => Bridge::DIRECTIONS, :uniqueness => { :scope => :table_id }

  delegate :start, :stop, :to => :table, :prefix => true
  delegate :name, :to => :user

  after_create :table_start
  after_destroy :table_stop
end

File: app/assets/javascripts/bridge/model/player/player.js
YUI.add("player-model", function (Y) {

    var Player = Y.Base.create("player", Y.Model, [], {

        list: function () {
            return this.lists[0];
        },

        table: function () {
            return this.list() && this.list().table();
        }

    }, {

        ATTRS: {

            name: {

            },

            direction: {

            }

        }

    });

    Y.namespace("Bridge.Model").Player = Player;

}, "", { requires: ["model"] });
